src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -10,35 +10,42 @@ const CanonicalChartbeatBeacon = ({
   chartbeatUID,
   useCanonical,
   chartbeatSource,
-}) => (
-  <Helmet>
-    <script async type="text/javascript">
-      {`
+  title,
+  referrer,
+}) => {
+  const chartbeatConfig = {
+    uid: chartbeatUID,
+    domain,
+    useCanonical,
+    useCanonicalDomain: useCanonical,
+    title,
+    type,
+    sections,
+    ...(!!referrer && { virtualReferrer: referrer }),
+    ...(!!cookie && { idSync: { bbc_hid: cookie } }),
+  };
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.pSUPERFLY) {
+      window.pSUPERFLY.virtualPage(chartbeatConfig);
+    }
+  });
+  return (
+    <Helmet>
+      <script async type="text/javascript">
+        {`
         (function(){
           var _sf_async_config = window._sf_async_config = (window._sf_async_config || {});
-          _sf_async_config.uid = ${chartbeatUID};
-          _sf_async_config.domain = "${domain}";
-          _sf_async_config.useCanonical = ${useCanonical};
-          _sf_async_config.useCanonicalDomain = ${useCanonical};
-          _sf_async_config.contentType = "${type}";
-          _sf_async_config.sections = "${sections}";
-          _sf_async_config.idSync = {
-           bbc_hid: "${cookie}"
-          };
-          function loadChartbeat() {
-           var e = document.createElement('script');
-           var n = document.getElementsByTagName('script')[0];
-           e.type = 'text/javascript';
-           e.async = true;
-           e.src = "${chartbeatSource}";
-           n.parentNode.insertBefore(e, n);
-          }
-          loadChartbeat();
+          _sf_async_config = Object.assign(_sf_async_config, ${JSON.stringify(
```

Comment:
https://caniuse.com/#search=object.assign. We might not be able to use `Object.assign` because we need to support the Opera Mini browser.

Reply:
I'm assuming this is because babel cannot transpile this piece of code as it's an interpolated string.

Reply:
thanks, @HarveyPeachey

Reply:
Can we avoid using assign completely and just assign the stringified JSON to the variables on L38?